lib/server/server-google-pubsub.ts

```diff
@@ -286,6 +308,14 @@ export class GooglePubSubTransport extends Server implements CustomTransportStra
         ];
     };
 
+    private async handleMessageSync([pattern, iterator]: [
+        string,
+        AsyncGenerator<GooglePubSubMessage>,
+    ]) {
+        const message = await iterator.next();
```

Comment:
This feels odd - doesn't this need to be in a `for await...of` loop?

Reply:
Also, what if the `message` here is `done`?

Reply:
good point, I switched to `for await..of` which also automatically handles `done` iterables

Reply:
But this raises a good point, if our subscription's message queue ever runs dry, we're going to unsubscribe from the listener and end the loop, right?

Reply:
It is, for all incense and porpoises, an infinite loop - you'll only leave if you error out or use a keyword like `return` or `break`